regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,197 @@
+                       )
+    subpi.add_argument("--cppflag", help="Current and pending form (true or false)"
+                       )
+    subpi.add_argument("--other_agencies", help="Other agencies to which the proposal has been "
+                                                "submitted"
+                       )
+    subpi.add_argument("--institution", help="The institution where the work will primarily"
+                                             "be carried out"
+                       )
+    subpi.add_argument("--months_academic", help="Number of working months in the academic year"
+                                                 "to be stated on the current and pending form"
+                       )
+    subpi.add_argument("--months_summer", help="Number of working months in the summer to be"
+                                                "stated on the current and pending form"
+                       )
+    subpi.add_argument("--scope", help="Scope of the project"
+                       )
+    subpi.add_argument("-f", "--funder",
+                       help="Agency where the proposal is being submitted"
+                       )
+    subpi.add_argument("-n", "--notes", nargs="+",
+                       help="Anything to note"
+                       )
+    return subpi
+
+
+class ProposalAdderHelper(DbHelperBase):
+    """Helper for adding a proposal to the proposals collection.
+
+       A proposal is a dictionary object describing a research or
+        project proposal submitted by the group.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_proposal"
+    needed_dbs = [f'{TARGET_COLL}', 'people']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        gtx = self.gtx
+        rc = self.rc
+        if not rc.date:
+            now = dt.date.today()
+        else:
+            now = date_parser.parse(rc.date).date()
+        key = f"{str(now.year)[2:]}_{''.join(rc.name.casefold().split()).strip()}"
+
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) > 0:
+            raise RuntimeError(
+                "This entry appears to already exist in the collection")
+        else:
+            pdoc = {}
+        pdoc.update({'_id': key})
+        if rc.amount:
+            pdoc.update({'amount': rc.amount})
+        else:
+            pdoc.update({'amount': ''})
```

Comment:
should be a number, not a string.  It may be better to have argparser to set the default values, I am not sure.  I know I have done it using this method too.  Maybe test it both ways, but it will make the code cleaner if the defaults from argparse works.

Reply:
Is this conditional in my code is redundant because `amount` is a required argument hence the code won't run if it is not specified? If this is the case, I will remove the `if` condition on all required arguments. 
Are you suggesting that I make `amount` an optional argument and set  `default = 0` for it in the `argparse`? In that case, if my understanding is correct:
1.  if the user specifies `--amount 50` the output `amount: 50` will be printed
2. if the user specifies only `--amount` the output `amount: 0` will be printed
3. if the user does not specify --amount`, no `amount` key-value pair will be printed

Reply:
yes, the first thing.